src/game/Laptop/IMP_SkillTraits.cc

```diff
@@ -0,0 +1,687 @@
+
+	for( uiCnt=0; uiCnt<IMP_SKILL_TRAITS__NUMBER_SKILLS; uiCnt++ )
+	{
+		usBoxPosX = usPosX - IMP_SKILL_TRAIT__GREY_BOX_OFFSET_X;
+		usBoxPosY = usPosY - IMP_SKILL_TRAIT__GREY_BOX_OFFSET_Y;
+
+		//if the merc is a FEMALE, skip this skill cause there isnt any fenmal martial artists
+		if( ShouldTraitBeSkipped( uiCnt ) )
+		{
+			//if its the none button
+			if( uiCnt == IMP_SKILL_TRAITS__NONE-1 )
+			{
+				usPosX = IMP_SKILL_TRAIT__NONE_BTN_LOC_X + IMP_SKILL_TRAIT__TEXT_OFFSET_X;
+				usPosY += IMP_SKILL_TRAIT__SPACE_BTN_BUTTONS;
+			}
+			continue;
+		}
+
+		//if the trait is selected
+		if( gfSkillTraitQuestions[ uiCnt ] )
+		{
+			//Display the gold background box
+			BltVideoObject(FRAME_BUFFER, guiIST_GreyGoldBox, 1, usBoxPosX, usBoxPosY);
+		}
+		else
+		{
+			//Display the grey background box
+			BltVideoObject(FRAME_BUFFER, guiIST_GreyGoldBox, 0, usBoxPosX, usBoxPosY);
+		}
+
+		//draw the text to the screenx
+		DrawTextToScreen(gzIMPSkillTraitsText[uiCnt], usPosX, usPosY, 0, IMP_SKILL_TRAIT__FONT, IMP_SKILL_TRAIT__COLOR, FONT_MCOLOR_BLACK, LEFT_JUSTIFIED);
+
+		//Determine the next x location
+		if( uiCnt < IMP_SKILL_TRAIT__SKILL_TRAIT_TO_START_RIGHT_COL )
+			usPosX = IMP_SKILL_TRAIT__LEFT_COLUMN_START_X + IMP_SKILL_TRAIT__TEXT_OFFSET_X;
+		else
+			usPosX = IMP_SKILL_TRAIT__RIGHT_COLUMN_START_X + IMP_SKILL_TRAIT__TEXT_OFFSET_X;
+
+		//Determine the next Y location
+		if( uiCnt == IMP_SKILL_TRAIT__SKILL_TRAIT_TO_START_RIGHT_COL )
+			usPosY = IMP_SKILL_TRAIT__RIGHT_COLUMN_START_Y + IMP_SKILL_TRAIT__TEXT_OFFSET_Y;
+		else
+			usPosY += IMP_SKILL_TRAIT__SPACE_BTN_BUTTONS;
+
+		//if its the none button
+		if( uiCnt == IMP_SKILL_TRAITS__NONE-1 )
+		{
+			usPosX = IMP_SKILL_TRAIT__NONE_BTN_LOC_X + IMP_SKILL_TRAIT__TEXT_OFFSET_X;
+		}
+	}
+}
+
+
+void BtnIMPSkillTraitFinishCallback(GUI_BUTTON *btn,INT32 reason)
+{
+	// btn callback for IMP personality quiz answer button
+	if (!(btn->uiFlags & BUTTON_ENABLED))
+		return;
+
+	if( reason & MSYS_CALLBACK_REASON_LBUTTON_DWN )
+	{
+		btn->uiFlags|=(BUTTON_CLICKED_ON);
+
+		//if we are just reviewing the page
+		if( iCurrentProfileMode == 5 )
+		{
+			//go back tot he done screen
+			iCurrentImpPage = IMP_FINISH;
+		}
+		else
+		{
+			iCurrentImpPage = IMP_MAIN_PAGE;
+
+			if( CameBackToSpecialtiesPageButNotFinished() )
+			{
+			}
+			else
+			{
+				//We are finished on this page
+				iCurrentProfileMode = 2;
+			}
+		}
+	}
+}
+
+
+BOOLEAN CameBackToSpecialtiesPageButNotFinished()
+{
+	//if we are in a page that comes after this one
+	if( iCurrentProfileMode == 2 )
+	{
+		return( TRUE );
+	}
+	else
+	{
+		return( FALSE );
+	}
+}
+
+INT8	DoesPlayerHaveExtraAttibutePointsToDistributeBasedOnSkillSelection()
+{
+	INT8		bExtraPoints=10;
+	INT8		bNumSkills=0;
+
+	//Count the number of skills selected
+	bNumSkills = CountNumSkillStraitsSelected( FALSE );
+
+	//if there NONE selected
+	if( bNumSkills == 0 )
+	{
+		bExtraPoints = 10;
+	}
```

Comment:
This block or the immediate initialization to 10 is not needed.

Reply:
Eh, nevermind, still haven't got used to this exceedingly verbose style. Cleanest to not init the var at all.